sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/MethodOverloadsShouldBeGroupedBase.cs

```diff
@@ -89,11 +90,70 @@ protected List<SyntaxToken>[] GetMisplacedOverloads(SyntaxNodeAnalysisContext c,
                 }
             }
             return misplacedOverloads.Values.Where(x => x.Count > 1).ToArray();
+
+            bool ProcessByInterface(TMemberDeclarationSyntax member, List<MemberInfo> others)
```

Comment:
IsMisplaced?

Reply:
If IsMisplaced or not is complex question answered by the entire `foreach`, variable state and bunch of `if`. This answers question if it ShouldBeProcessed based on two different interface criteria (not only interface grouping).
`ShouldProcess` ?

Reply:
IsMisplacedCandidate? I think that Process is too generic and it doesn't provide additional value